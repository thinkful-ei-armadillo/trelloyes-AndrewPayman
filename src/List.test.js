import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

describe('List component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    //render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(
      <List
        header="test list"
        key={2}
        cards={[
          { id: 'g', title: 'First card', content: 'lorem ipsum' },
          { id: 'f', title: 'Second card', content: 'lorem ipsum' }
        ]}
      />,
      div
    );

    //clean up code
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <List
          header="test list"
          key={1}
          cards={[
            { id: 'a', title: 'First card', content: 'lorem ipsum' },
            { id: 'b', title: 'Second card', content: 'lorem ipsum' }
          ]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
