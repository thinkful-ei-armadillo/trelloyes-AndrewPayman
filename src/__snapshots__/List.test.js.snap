// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component testing renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      test list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
